imageio/core/util.py

```diff
@@ -119,7 +119,10 @@ def image_as_uint(im, bitdepth=None):
         if not np.isfinite(ma):
             raise ValueError("Maximum image value is not finite")
         if ma == mi:
-            raise ValueError("Max value == min value, ambiguous given dtype")
+            if np.all(im == ma): 
```

Comment:
This can be `if mi == ma` (which would probably be better when there are NaNs).

Reply:
Ok I will change it so that it doesn't raise an error if mi == ma and just returns im.astype(out_type)